src/modules/timepix3/test/Timepix3ParserTest.cpp

```diff
@@ -91,7 +100,13 @@ TEST_F(Timepix3ParserTest, TDCReadouts) {
   EXPECT_EQ(counters.TDC1FallingReadouts, 0);
   EXPECT_EQ(counters.TDC2RisingReadouts, 0);
   EXPECT_EQ(counters.TDC2FallingReadouts, 0);
- 
+
+  EXPECT_TRUE(testEventHandler.getLastTdcEvent() != nullptr);
+  EXPECT_TRUE(testEventHandler.getLastTdcEvent()->Type == 15);
+  EXPECT_TRUE(testEventHandler.getLastTdcEvent()->TriggerCounter == 2231);
+  EXPECT_TRUE(testEventHandler.getLastTdcEvent()->Timestamp == 31447764897);
+  EXPECT_TRUE(testEventHandler.getLastTdcEvent()->Stamp == 6);
```

Comment:
With the new pattern, I was able to improve and significantly test the parser is doing the correct job not just checking that counters are increased.